src/v/model/metadata.h

```diff
@@ -468,7 +468,8 @@ enum class cloud_storage_backend {
     google_s3_compat = 1,
     azure = 2,
     minio = 3,
-    unknown = 4,
+    oracle_s3_compat = 4,
+    unknown
```

Comment:
are names or enum numeric values stored in the configuration system? if the later then it seems problematic to assign a different enum to the `4` value.

Reply:
[The `enum` type](https://github.com/redpanda-data/redpanda/blob/10eb41eabdd08e7b9d9154f4c0d5c25abd20bd2a/src/v/config/property.h#L819-L820) is used to template `property<T>`,  which also uses the [`enum` type and not the `underlying_type`](https://github.com/redpanda-data/redpanda/blob/10eb41eabdd08e7b9d9154f4c0d5c25abd20bd2a/src/v/config/property.h#L74-L79).

Reply:
what i mean is: if the underlying numeric enum value is stored in the controller log, then it'd be a problem. does the controller log store the stringified name or the underlying numeric vavlue?

Reply:
[https://github.com/redpanda-data/redpanda/blob/c4149595816695976050a92148f0f741d6a8c90e/src/v/cluster/types.h#L1456-L1467](From what I am reading, keys and values are left in raw string format when applying updates in the `config_manager` state machine)

I now see the reason for concern though. Persisting this configuration in `raft0` as a the raw `underlying_type` didn't cross my mind when making this change- I'll keep staring at it and revert the change if it ends up being dangerous.

Reply:
thanks!